core/sawtooth/client.py

```diff
@@ -779,24 +779,34 @@ def get_transaction_status(self, txnid, timeout=30):
             request.get_method = lambda: 'HEAD'
             opener = urllib2.build_opener(self.proxy_handler)
             response = opener.open(request, timeout=timeout)
-            code = response.getcode()
-            response.close()
-            if code == 200:
-                return transaction.Status.committed
-            elif code == 302:
-                return transaction.Status.pending
-            else:
-                return transaction.Status.unknown
 
         except urllib2.HTTPError as err:
             LOGGER.error('peer operation on url %s failed with response: %d',
                          url, err.code)
+            if err.code == 302:
```

Comment:
If the normal flow is that a 302 does not generate a HTTPError exception, I'm still wondering why we need to check for 302 here.  Do they actually both mean pending-with-no-error?